### PR TITLE
Fix reveal background

### LINKED ISSUE #8377: Hover Reveal style from Dalston theme not fading the Cover layer
### Quick summary

While working on https://github.com/Automattic/themes/pull/8376, I noticed that https://github.com/Automattic/themes/pull/1638 wasn't working.

It should have an effect in the Cover layer similar to the effect in the content being revealed.

Not sure if it's related to an old version of the Cover block, but currently the class `.has-background-dim` is not applied to the same element that has the class `.wp-block-cover`. It's a child element.

### Steps to reproduce

1. Create a cover block with the "Hover Reveal" style.
2. Add some content.
3. Add a gradient and opacity.
4. Check that only the content is hidden/revealed.

### A clear and concise description of what you expected to happen.

Based on the code, and on the https://github.com/Automattic/themes/pull/1638, it should have an effect in the layer before the content either.

### What actually happened

The effect is happening only in the content and not in the layer.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### If the above answer is "Yes...", outline the workaround.

_No response_

### Platform (Simple and/or Atomic)

_No response_

### Logs or notes

- Theme: Dalston
- When working on it, also keep in mind the touch devices with the `dalson-in-viewport` classes. See https://github.com/Automattic/themes/pull/8376/files#r1831286076.

### PR DESCRIPTION
**Pull request by renatho**
#### Changes proposed in this Pull Request:

It fixes the Hover Reveal style of the Cover block for the latest versions of WordPress.

#### Known issues:

- https://github.com/Automattic/themes/issues/8386
- https://github.com/Automattic/themes/issues/8395

#### Test instructions:

I tested upgrading to all major releases from 5.0 (5.1, 5.2, 5.3...). And I tried to simplify it getting some specific versions. But feel free to test bumping 1 in 1 if you have time. :)

For every update, make sure you update the blocks in the editor by editing anything in the page and clicking on "Update" to get the correct version of the blocks.

In some updates it will also ask for a database update. Confirm it when requested.

- Setup an environment with WP 5.0 (first version with Gutenberg).
  - Create a page with some cover blocks with some variations:
    - With image and background.
    - For each (image and background), use the different options (using the default and changing the color overlay and opacity).
  - Set the style "Hover Reveal" to the blocks.
  - Check that it works properly when you hover the elements.
- Update your WP to 5.2.
  - Update the blocks
  - Make sure everything continues working well.
- Update your WP to 5.3.
  - Create a cover block only with color (without image or video) and the Hover Reveal style.
  - Make sure everything continues working well.
- Update your WP to 5.4.
  - Add variations with gradient, using video, image, and only text.
  - Use a different opacity in one of them.
  - Make sure everything continues working well.
- Update your WP to 5.5.
  - Update the blocks.
  - See that everything continues working, but you will see an issue when hovering the text in the editor. I created [an issue](https://github.com/Automattic/themes/issues/8386) for it.
- Upgrade to WP 5.9, and see that the background will always be visible, regardless if we are in this branch or trunk. I tried to fix it, but it was complex given all possible scenarios and I was also afraid to break for other versions, so I just created [an issue](https://github.com/Automattic/themes/issues/8395) for now since the more important would be to fix it for the latest versions.
- Update to WP 6.0.
  - Make sure everything continues working well.
- Update to WP 6.3.
  - Make sure everything continues working well.
- Update to WP 6.6.
  - Make sure everything continues working well.

https://github.com/user-attachments/assets/ee7d50f8-6775-4776-a242-8230439e2607

#### Related issue(s):

https://github.com/Automattic/themes/issues/8377

### CODE REVIEW

File: dalston/block-extends/extend-cover-block.css [hunk 1]
```diff
@@ -144,23 +144,27 @@
 	text-decoration: underline;
 }
 
-.wp-block-cover.is-style-hover-reveal .wp-block-cover__inner-container {
+.wp-block-cover.is-style-hover-reveal .wp-block-cover__inner-container,
+.wp-block-cover.is-style-hover-reveal .wp-block-cover-text {
```
Comment (renatho) [inline dalston/block-extends/extend-cover-block.css:148]: The addition of `.wp-block-cover-text` is to fix the reveal in old versions of Gutenberg where it didn't have the InnerBlocks in the Cover yet.

File: dalston/block-extends/extend-cover-block.css [hunk 2]
```diff
@@ -2,10 +2,10 @@
  * Editor
  */
 
-#editor .wp-block-cover-image.is-style-hover-reveal.has-background-dim.has-background-dim .wp-block-cover__gradient-background,
```
Comment (renatho) [inline dalston/block-extends/extend-cover-block.css:5]: Just remove the not needed duplicate selector `.has-background-dim.has-background-dim`.

File: dalston/block-extends/extend-cover-block.css [hunk 3]
```diff
@@ -144,23 +144,27 @@
...
 	opacity: 1;
 }
 
 /* Dim */
 
 .wp-block-cover.is-style-hover-reveal.has-background-dim:before,
 .wp-block-cover.is-style-hover-reveal.has-background-dim:not(.has-background-gradient):before,
-.wp-block-cover.is-style-hover-reveal.has-background-dim.has-background-gradient .wp-block-cover__gradient-background {
+.wp-block-cover.is-style-hover-reveal.has-background-dim.has-background-gradient .wp-block-cover__gradient-background,
+.wp-block-cover.is-style-hover-reveal .wp-block-cover__background.has-background-dim {
```
Comment (renatho) [inline dalston/block-extends/extend-cover-block.css:167]: In the current Gutenberg version the `has-background-dim` class is applied to the element with `wp-block-cover__background` class and not to the `wp-block-cover`.

File: dalston/block-extends/extend-cover-block.css [hunk 4]
```diff
@@ -169,81 +173,105 @@
 
 .wp-block-cover.is-style-hover-reveal.dalston-in-viewport.has-background-dim.has-background-dim-0:before,
 .wp-block-cover.is-style-hover-reveal.dalston-in-viewport.has-background-dim.has-background-dim-0.has-background-gradient .wp-block-cover__gradient-background,
+.wp-block-cover.is-style-hover-reveal.dalston-in-viewport .wp-block-cover__background.has-background-dim.has-background-dim-0,
 .wp-block-cover.is-style-hover-reveal.has-background-dim.has-background-dim-0:hover:before,
-.wp-block-cover.is-style-hover-reveal.has-background-dim.has-background-dim-0.has-background-gradient:hover .wp-block-cover__gradient-background {
+.wp-block-cover.is-style-hover-reveal.has-background-dim.has-background-dim-0.has-background-gradient:hover .wp-block-cover__gradient-background,
+.wp-block-cover.is-style-hover-reveal:hover .wp-block-cover__background.has-background-dim.has-background-dim-0 {
```
Comment (renatho) [inline dalston/block-extends/extend-cover-block.css:179]: In the current Gutenberg version the `has-background-dim` class is applied to the element `.wp-block-cover__background` and not to the `.wp-block-cover`.

File: dalston/block-extends/extend-cover-block.css [hunk 5]
```diff
@@ -144,23 +144,27 @@
...
 	opacity: 1;
 }
 
 /* Dim */
 
 .wp-block-cover.is-style-hover-reveal.has-background-dim:before,
 .wp-block-cover.is-style-hover-reveal.has-background-dim:not(.has-background-gradient):before,
-.wp-block-cover.is-style-hover-reveal.has-background-dim.has-background-gradient .wp-block-cover__gradient-background {
+.wp-block-cover.is-style-hover-reveal.has-background-dim.has-background-gradient .wp-block-cover__gradient-background,
+.wp-block-cover.is-style-hover-reveal:has(.wp-block-cover__image-background, .wp-block-cover__video-background) .wp-block-cover__background.has-background-dim {
```
Comment (renatho) [inline dalston/block-extends/extend-cover-block.css:167]: Here, with the `:has(.wp-block-cover__image-background, .wp-block-cover__video-background)`, we make sure the overlay will be hidden only when it has an image or video.

**Review by phcp — APPROVED** [review]
Nice work! I tested it on almost all suggested versions (except 5.2), and it works as expected. I also spotted the errors mentioned in the instructions for versions 5.5 and 5.9. For 5.2, I got a `TypeError: Cannot read properties of null (reading 'innerBlocks')`, which seems unrelated to the changes in this PR. 
p1731092353995719/1731073822.997419-slack-C07GZ2UA3TN

**Pull request merged**